src/main/java/de/tum/in/www1/artemis/web/rest/tutorialgroups/TutorialGroupResource.java

```diff
@@ -0,0 +1,279 @@
+        var tutorialGroupFromDatabase = this.tutorialGroupRepository.findByIdWithTeachingAssistantAndRegistrationsElseThrow(tutorialGroupId);
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, tutorialGroupFromDatabase.getCourse(), null);
+        tutorialGroupRepository.deleteById(tutorialGroupFromDatabase.getId());
+        return ResponseEntity.noContent().build();
+    }
+
+    /**
+     * PUT /tutorial-groups : Updates an existing tutorial group
+     *
+     * @param tutorialGroup group the tutorial group to update
+     * @return the ResponseEntity with status 200 (OK) and with body the updated tutorial group
+     */
+    @PutMapping("/tutorial-groups")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    @FeatureToggle(Feature.TutorialGroups)
+    public ResponseEntity<TutorialGroup> update(@RequestBody @Valid TutorialGroup tutorialGroup) {
+        log.debug("REST request to update TutorialGroup : {}", tutorialGroup);
+        if (tutorialGroup.getId() == null) {
+            throw new BadRequestException("A tutorial group cannot be updated without an id");
+        }
+        var tutorialGroupFromDatabase = this.tutorialGroupRepository.findByIdWithTeachingAssistantAndRegistrationsElseThrow(tutorialGroup.getId());
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, tutorialGroupFromDatabase.getCourse(), null);
+
+        trimStringFields(tutorialGroup);
+        if (!tutorialGroupFromDatabase.getTitle().equals(tutorialGroup.getTitle())) {
+            checkTitleIsUnique(tutorialGroup);
+        }
+        overrideValues(tutorialGroup, tutorialGroupFromDatabase);
+
+        var updatedTutorialGroup = tutorialGroupRepository.save(tutorialGroupFromDatabase);
+        return ResponseEntity.ok(updatedTutorialGroup);
+    }
+
+    /**
+     * DELETE tutorial-groups/:tutorialGroupId/deregister/:studentLogin : deregister a student from a tutorial group.
+     *
+     * @param tutorialGroupId the id of the tutorial group
+     * @param studentLogin    the login of the student to deregister
+     * @return the ResponseEntity with status 204 (NO_CONTENT)
+     */
+    @DeleteMapping("/tutorial-groups/{tutorialGroupId}/deregister/{studentLogin:" + Constants.LOGIN_REGEX + "}")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    @FeatureToggle(Feature.TutorialGroups)
+    public ResponseEntity<Void> deregisterStudent(@PathVariable Long tutorialGroupId, @PathVariable String studentLogin) {
+        log.debug("REST request to deregister {} student from tutorial group : {}", studentLogin, tutorialGroupId);
+        var tutorialGroupFromDatabase = this.tutorialGroupRepository.findByIdElseThrow(tutorialGroupId);
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, tutorialGroupFromDatabase.getCourse(), null);
+
+        Optional<User> studentToDeregister = userRepository.findOneWithGroupsAndAuthoritiesByLogin(studentLogin);
```

Comment:
You could use `userRepository.getUserWithGroupsAndAuthorities()`, as this already has the isEmpty check with the correct `EntityNotFoundException` included.

Reply:
Good idea! Done in 282a84a2a83ea0d17d0d693e02b3114c978652d5